### PR TITLE
Remove corefx patches

### PR DESCRIPTION
**Pull request by weshaggard**
cc @eerhardt

### CODE REVIEW

File: targets/corefx.props [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup>
     <ProjectDirectory Condition="'$(PathToRepo)' != ''">$(PathToRepo)</ProjectDirectory>
-    <BuildCommand>$(ProjectDirectory)/build$(ShellExtension) -$(Configuration) -buildArch=$(Platform) -portable=false -- /p:ILLinkTrimAssembly=false /p:BuildAllPackages=true</BuildCommand>
+    <BuildCommand>$(ProjectDirectory)/build$(ShellExtension) -$(Configuration) -buildArch=$(Platform) -portable=false -allConfigurations -- /p:ILLinkTrimAssembly=false</BuildCommand>
```
Comment (eerhardt) [inline targets/corefx.props]: To me, it seems like corefx should be checking the `$(DotNetBuildFromSource)` flag here, instead of source-build having to specify `-allConfigurations`.

Do you plan on doing this later?

File: targets/corefx.props [hunk 2]
```diff
@@ -14,11 +14,13 @@
...
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(OfflineBuild)' == 'true'">
     <EnvironmentVariables Include="LatestCommit=$(LatestCommit)" />
     <EnvironmentVariables Include="OfficialBuildId=$(OfficialBuildId)" />
     <EnvironmentVariables Include="VersionSeedDate=$(OfficialBuildId)" />
-    <EnvironmentVariables Include="OverridePackageSource=$(SourceBuiltPackagesPath)" />
-    <EnvironmentVariables Include="OverridePackageSourceEx=$(PrebuiltPackagesPath)" Condition="'$(OfflineBuild)' == 'true'" />
-    <EnvironmentVariables Include="ExcludeInternetFeeds=true" Condition="'$(OfflineBuild)' == 'true'" />
+    <EnvironmentVariables Include="DotNetSourceBuildRestoreSources=$(SourceBuiltPackagesPath)%3B$(PrebuiltPackagesPath)" />
```
Comment (eerhardt) [inline targets/corefx.props]: This is duplicated in `targets/standard.props`.  Should it be move to a common place?

File: targets/corefx.props [hunk 3]
```diff
@@ -14,11 +14,13 @@
   <ItemGroup>
     <RepositoryReference Include="coreclr" />
     <RepositoryReference Include="standard" />
+    <EnvironmentVariables Include="DotNetSourceBuildRestoreSources=$(SourceBuiltPackagesPath)" Condition="'$(OfflineBuild)' != 'true'" />
+  </ItemGroup>
+
+  <ItemGroup Condition="'$(OfflineBuild)' == 'true'">
```
Comment (eerhardt) [inline targets/corefx.props]: I think we want the `LatestCommit` `OfficialBuildId` `VersionSeedDate` parameters even if we aren't building offline.  Why the change?

Comment (weshaggard) [reply targets/corefx.props]: They are only interesting when building from the tarball. For the LatestCommit we will get the live one from git and for OfficialBuildId/VersionSeedDate we only want that if we are producing an official build and we need it to exactly match one we shipped which again is only interesting for the tar-ball.

File: targets/corefx.props (hunk 1 above)
Comment (weshaggard) [reply targets/corefx.props]: I agree and plan to do it later, when we start to solidify the exact command lines we pass to the repos.

File: targets/corefx.props (hunk 2 above)
Comment (weshaggard) [reply targets/corefx.props]: Yes it probably is and I will move these into the common place as part of my coreclr work.

**Comment by weshaggard** [comment]
test RHEL7.2 Release

**Comment by weshaggard** [comment]
test RHEL7.2 Release

File: targets/corefx.props [hunk 4]
```diff
@@ -14,11 +14,13 @@
   <ItemGroup>
     <RepositoryReference Include="coreclr" />
     <RepositoryReference Include="standard" />
+    <EnvironmentVariables Include="DotNetSourceBuildRestoreSources=$(SourceBuiltPackagesPath)" Condition="'$(OfflineBuild)' != 'true'" />
```
Comment (dagood) [inline targets/corefx.props]: To make sure my plan is known: I'm planning on removing this and using the [`DotNetRestoreSourcePropsPath`](https://github.com/dotnet/source-build/blob/dev/release/2.0/Documentation/auto-dependency-flow/api.md#-pdotnetrestoresourcepropspathpath) Repo API when I have my changes through.

Re: https://gitter.im/dotnet/source-build?at=59d29381614889d47567f95b
@eerhardt @weshaggard

Comment (weshaggard) [reply targets/corefx.props]: Yep.

File: targets/corefx.props (hunk 1 above)
Comment (eerhardt) [reply targets/corefx.props]: Since this is a servicing branch, don't you want `/p:BuildAllPackages=true`?  Or does `-allConfigurations` do the same thing?

**Comment by weshaggard** [comment]
test CentOS7.1 Release
test RHEL7.2 Release

**Comment by weshaggard** [comment]
@eerhardt corefx patches are green on RHEL now so I"m going to merge this finally.
